FirebaseRemoteConfig/Tests/Unit/RCNInstanceIDTest.m

```diff
@@ -0,0 +1,400 @@
+                      DBManager:(RCNConfigDBManager *)DBManager
+                       settings:(RCNConfigSettings *)settings
+                     experiment:(RCNConfigExperiment *)experiment
+                          queue:(dispatch_queue_t)queue
+                      namespace:firebaseNamespace
+                            app:firebaseApp;
+@end
+
+@interface FIRRemoteConfig (ForTest)
+- (void)updateWithNewInstancesForConfigFetch:(RCNConfigFetch *)configFetch
+                               configContent:(RCNConfigContent *)configContent
+                              configSettings:(RCNConfigSettings *)configSettings
+                            configExperiment:(RCNConfigExperiment *)configExperiment;
+@end
+
+@implementation FIRRemoteConfig (ForTest)
+- (void)updateWithNewInstancesForConfigFetch:(RCNConfigFetch *)configFetch
+                               configContent:(RCNConfigContent *)configContent
+                              configSettings:(RCNConfigSettings *)configSettings
+                            configExperiment:(RCNConfigExperiment *)configExperiment {
+  [self setValue:configFetch forKey:@"_configFetch"];
+  [self setValue:configContent forKey:@"_configContent"];
+  [self setValue:configSettings forKey:@"_settings"];
+  [self setValue:configExperiment forKey:@"_configExperiment"];
+}
+@end
+
+@interface RCNConfigDBManager (Test)
+- (void)removeDatabaseOnDatabaseQueueAtPath:(NSString *)path;
+@end
+
+@interface RCNUserDefaultsManager (Test)
++ (NSUserDefaults *)sharedUserDefaultsForBundleIdentifier:(NSString *)bundleIdentifier;
+@end
+
+typedef NS_ENUM(NSInteger, RCNTestRCInstance) {
+  RCNTestRCInstanceDefault,
+  RCNTestRCInstanceSecondNamespace,
+  RCNTestRCInstanceSecondApp,
+  RCNTestRCNumTotalInstances
+};
+
+@interface FIRInstanceID (Tests)
++ (FIRInstanceID *)instanceIDForTests;
+@end
+
+@interface RCNInstanceIDTest : XCTestCase {
+  NSTimeInterval _expectationTimeout;
+  NSTimeInterval _checkCompletionTimeout;
+  NSMutableArray<FIRRemoteConfig *> *_configInstances;
+  NSMutableArray<NSDictionary<NSString *, NSString *> *> *_entries;
+  NSMutableArray<NSDictionary<NSString *, id> *> *_response;
+  NSMutableArray<NSData *> *_responseData;
+  NSMutableArray<NSURLResponse *> *_URLResponse;
+  NSMutableArray<RCNConfigFetch *> *_configFetch;
+  RCNConfigDBManager *_DBManager;
+  NSUserDefaults *_userDefaults;
+  NSString *_userDefaultsSuiteName;
+  NSString *_DBPath;
+}
+@end
+
+@implementation RCNInstanceIDTest
+
+- (void)setUp {
+  [super setUp];
+  FIRSetLoggerLevel(FIRLoggerLevelMax);
+  if (![FIRApp isDefaultAppConfigured]) {
+    [FIRApp configureWithOptions:[self firstAppOptions]];
+  }
+  _expectationTimeout = 5;
+  _checkCompletionTimeout = 1.0;
+
+  // Always remove the database at the start of testing.
+  _DBPath = [RCNTestUtilities remoteConfigPathForTestDatabase];
+  id classMock = OCMClassMock([RCNConfigDBManager class]);
+  OCMStub([classMock remoteConfigPathForDatabase]).andReturn(_DBPath);
```

Comment:
Do we need to stop mocking the class at the end of the test to avoid interference with other tests? If so `classMock` should be stored to an ivar  and taken care in `tearDown` method.

The same for similar mocks below.

Reply:
Done.